### ISSUE TITLE
Supporting non-http/https pages

### ISSUE THREAD

**Issue by TheNamlessGuy** [Robbendebiene/Gesturefy#728]
Hi! I guess this is more of a feature request/support question than a bug report, so please excuse the fact that I don't use the provided template.

Today, I got [an issue](https://github.com/TheNamlessGuy/browser-tab-grouper/issues/6) opened for my plugin, saying that this plugin doesn't work on the tabs it generates. I saw that it's mentioned that `moz-extension` tabs aren't supported in [the trouble shooting guide](https://github.com/Robbendebiene/Gesturefy/wiki/Troubleshooting). I wondered why, and after looking into how the extension works, I guess it's due to [`<all_urls>`](https://github.com/Robbendebiene/Gesturefy/blob/master/src/manifest.json#L38) [not supporting it](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/Match_patterns)?

If that's the reason why, would it be possible to use a combination of [browser.tabs.onUpdated](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/tabs/onUpdated) and [browser.scripting.executeScript](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/scripting/executeScript)/[browser.scripting.insertCSS](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/scripting/insertCSS) in order to (perhaps optionally) also inject the relevant code into those types of pages? And even if `<all_urls>` isn't the reason why, would it be possible to go that route regardless?

EDIT: I could help look into this myself when I get some more time on my hands, if this is too big of an ask.

**Comment by Robbendebiene** [comment]
Hi, thanks for the detailed request.
As far as I know there is now way for an extension to run code in foreign `moz-extension` pages. This is an inherent security aspect of webextensions. So using `browser.scripting.executeScript`/`browser.scripting.insertCSS` should fail with an error.

Extensions can only communicate with each other through messages. So in theory we could manage to get this to work if Gesturefy sends a message to Tab Grouper containing the necessary code which Tab Grouper would have to call `eval()` on. However this would require you to basically replicate a complete Gesturefy instance, so this isn't anywhere near from feasible. Additionally the use of `eval` in Manifest v3 won't be supported.

Long story short, unfortunately we cannot solve this problem 🫣
